### PR TITLE
Issue 1296

### PR DESCRIPTION
**Pull request by dalabre**
The test results backing this PR is exhibited with Issue #1296 (now closed).

### CODE REVIEW

**Comment by ndellingwood** [comment]
@dalabre this PR needs to go to the develop branch of Kokkos instead of the master branch. I'll update if I have permissions...

**Comment by ndellingwood** [comment]
@dalabre I changed it to merge with the develop branch, for future reference if you click the Edit button you will see this option right below the title.

File: generate_makefile.bash [hunk 1]
```diff
@@ -206,6 +213,11 @@ KOKKOS_SETTINGS="KOKKOS_SRC_PATH=${KOKKOS_SRC_PATH}"
 
 if [ ${#COMPILER} -gt 0 ]; then
   KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
+elif
+   [ ${#COMPILER} -eq 0 ] && [ ${#KOKKOS_DEVICES} -gt 0 ]; then
+  COMPILER="${KOKKOS_PATH}/bin/nvcc_wrapper"                                                  
+  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"   
+  #echo "CXX= ${CXX}"
```
Comment (ndellingwood) [inline generate_makefile.bash]: @dalabre go ahead and delete this line.

**Comment by dalabre** [comment]
Changes made per review comments and committed to issue-1296 branch on dalabre/kokkos fork.

File: generate_makefile.bash [hunk 2]
```diff
@@ -206,6 +212,10 @@ KOKKOS_SETTINGS="KOKKOS_SRC_PATH=${KOKKOS_SRC_PATH}"
 
 if [ ${#COMPILER} -gt 0 ]; then
   KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
+elif
+   [ ${#COMPILER} -eq 0 ] && [ ${#KOKKOS_DEVICES} -gt 0 ]; then
```
Comment (crtrott) [inline generate_makefile.bash]: This doesn't check whether KOKKOS_DEVICES includes Cuda. It should only automatically change to nvcc_wrapper if COMPILER is not set and KOKKOS_DEVICES contains Cuda.

Comment (ndellingwood) [reply generate_makefile.bash]: @dalabre from our session sitting down, try something like:
```
if echo "${KOKKOS_DEVICES}" | grep -q "Cuda"; then
  COMPILER="${KOKKOS_PATH}/bin/nvcc_wrapper"                                                  
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"  
fi
```

Comment (dalabre) [reply generate_makefile.bash]: Testing fixes ....

Comment (dalabre) [reply generate_makefile.bash]: .... tweak compile setting (remove double quotes)
```
      COMPILER=${COMPDIR}/${COMPNAME}
```
and add check for Cuda device before setting Cuda compiler
```
if [ ${#COMPILER} -gt 0 ]; then
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
elif
   [ ${#COMPILER} -eq 0 ] && [[ ${KOKKOS_DEVICES} =~ .*Cuda.* ]]; then
  COMPILER="${KOKKOS_PATH}/bin/nvcc_wrapper"
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"   
fi
```
Note, this is the string-equivalent of the file-capabable *grep -q 'string' $file*; the double [[ ]]
are required for this implementation.

changes pushed to dalabre/kokkos

**Comment by dalabre** [comment]
Testing of updated generate_makefile.bash completed; all behaviors as expected (pass/fail according to modules loaded and options specified or not specified on command line).

Please review.

**Pull request merged**